core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -453,7 +453,31 @@ contract PricelessPositionManager is FeePayer {
         oracle.requestPrice(priceIdentifer, requestedTime);
     }
 
-    function _getOraclePrice(uint requestedTime) internal view returns (FixedPoint.Unsigned memory) {
+    function _payFinalFee() internal {
+        // Send the fee payment.
+        FixedPoint.Unsigned memory totalPaid = super.payFinalFees();
+
+        // Exit early if totalCollaterall == 0 to prevent divide by 0.
+        if (totalPositionCollateral.isEqual(FixedPoint.fromUnscaledUint(0))) {
+            return;
+        }
+
+        // TODO(#873): add divCeil and mulCeil to make sure that all rounding favors the contract rather than the user.
+        // Adjust internal variables below.
+        // Compute fee percentage that was paid by the entire contract (fees / collateral).
+        // Unlike payFees, we are spreading fees across all locked collateral and NOT all PfC, which
+        // implies that we are not forcing liquidations to be responsible for paying final fees when the position expires
+        FixedPoint.Unsigned memory feePercentage = totalPaid.div(totalPositionCollateral);
+
+        // Compute adjustment to be applied to the position collateral (1 - feePercentage).
+        FixedPoint.Unsigned memory adjustment = FixedPoint.fromUnscaledUint(1).sub(feePercentage);
+
+        // Apply fee percentage to adjust totalPositionCollateral and positionFeeAdjustment.
+        totalPositionCollateral = totalPositionCollateral.mul(adjustment);
+        positionFeeAdjustment = positionFeeAdjustment.mul(adjustment);
+    }
+
+    function _getOraclePrice(uint requestedTime) public view returns (FixedPoint.Unsigned memory) {
```

Comment:
Did you mean to change this to `public`?

Reply:
ah nope good catch

Reply:
this is now (back to) `internal`